kedro/framework/cli/starters.py

```diff
@@ -143,6 +144,27 @@ class KedroStarterSpec:  # noqa: too-few-public-methods
     "7": "Kedro Viz",
 }
 
+VALIDATION_PATTERNS = {
+    "yes_no": {
+        "regex": r"(?i)^\s*(y|yes|n|no)\s*$",
+        "error_message": "|It must contain only y, n, YES, NO, case insensitive.",
+    }
+}
+
+
+def _validate_regex(pattern_name, text):
```

Comment:
`_validate_regex` is maybe a bit of a misleading name, because it sounds like you can use this generically for regex patterns, but it's actually quite specific for the yes/no choice. I'd rename it to something like `_validate_yes_no` or `_validate_example_choice`

Reply:
hm, the idea was that it's a general func to validate any regex pattern, and we should use it also for other regex validations, that currently hardcoded and duplicated in starters.py

Reply:
Ahh I see now. So the idea is to also expand `VALIDATION_PATTERNS` to contain other patterns than yes/no? In that case ignore my comment :)